Automattic/wp-calypso
# composite-checkout: Clean up unnecessary exports

## Proposed Changes

This PR aims to remove unused and unnecessary exports from the `@automattic/composite-checkout` package. The package has gone through many iterations over the years and has gained many exposed functions and components, but many of those no longer fit the vision of the package in its current form. Some can be moved out of the package to where they are used and some can be removed entirely.

To do:

- [x] Remove unused exports: CheckoutOrderSummary, CheckoutOrderSummaryStep, CheckoutOrderSummaryStepTitle, CheckoutStepAreaWrapper, InvalidPaymentProcessorResponseError, MainContentWrapper, OrderReviewLineItems, OrderReviewSection, OrderReviewTotal, RadioButton, getDefaultOrderReviewStep, getDefaultOrderSummary, getDefaultOrderSummaryStep, getDefaultPaymentMethodStep, useLineItems, useLineItemsOfType.
- [x] Refactor code to remove CheckoutCheckIcon, CheckoutSubmitButton, CheckoutSummaryArea, CheckoutSummaryCard, LoadingContent, PaymentLogo, SubmitButtonWrapper.

Some things that would be nice to remove but are outside the scope of this PR: 

- CheckoutPaymentMethods
- Button
- useTotal (and the whole line items API)

This mostly doesn't make any visible changes except for the following.

Payment method selector before:

<img width="765" alt="Screenshot 2023-09-21 at 11 46 28 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/7fee2acc-f9ea-4ffb-9594-5359cf3f51ca">

Payment method selector after:

<img width="762" alt="Screenshot 2023-09-21 at 11 46 32 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/8d113cf8-806c-4454-afac-158889f00731">

## Testing Instructions

Automated tests and TypeScript will cover nearly everything.

A few things worth testing manually:

- Make sure that the loading page shows normally when checkout is loading. Then decrease `showLoadingInfoThresholdMs` in `CheckoutLoadingPlaceholder` to 100 and make sure you see the extra loading content appear above the normal loading content while checkout is loading.
- Make sure the "Included with your purchase" section looks the same before/after this PR.
- Visit the assign new payment method page by visiting `/me/purchases` on a site with at least one subscription. Click on the subscription and then click "Change payment method" or "Add payment method". Verify that it looks mostly the same before and after this PR, except that the submit button is slightly different (see screenshots above).
